rt/transports/jms/src/main/java/org/apache/cxf/transport/jms/JMSConduit.java

```diff
@@ -511,6 +511,7 @@ private synchronized void shutdownListeners() {
     }
     public synchronized void close() {
         shutdownListeners();
+        jmsConfig.resetCachedReplyDestination();
```

Comment:
One similar modification is need in `sendExchange` method, right now the connection is closed before the `jmsConfig.resetCachedReplyDestination();`, so there is no chance to delete temporary query. The correct sequence should be:

```
                jmsConfig.resetCachedReplyDestination();
                ResourceCloser.close(connection);
                this.connection = null;
```

Reply:
You're right, done.